source/extensions/transport_sockets/tls/context_config_impl.cc

```diff
@@ -179,7 +179,15 @@ ContextConfigImpl::ContextConfigImpl(
                                                 default_min_protocol_version)),
       max_protocol_version_(tlsVersionFromProto(config.tls_params().tls_maximum_protocol_version(),
                                                 default_max_protocol_version)),
-      handshaker_factory_(*HandshakerFactoryImpl::getDefaultHandshakerFactory()) {
+      handshaker_factory_cb_([&]() {
+        auto* factory = HandshakerFactoryImpl::getDefaultHandshakerFactory();
+        auto handshaker_factory_context = HandshakerFactoryContextImpl(api_, alpnProtocols());
```

Comment:
style: `HandshakerFactoryContextImpl handshaker_factory_context(api_, alpnProtocols());`

Reply:
and this object should be shared betwen both the config-provided and default case (ie put it before the if/else from previous comment)

Reply:
Done.